File: geobus-backend/src/main/java/com/geobus/marrakech/service/BusPositionService.java
package com.geobus.marrakech.service;

import java.time.LocalDateTime;
import java.util.List;

import com.geobus.marrakech.model.BusPosition;

/**
 * Interface pour les services liés aux positions des bus
 */
public interface BusPositionService {

    /**
     * Récupère les positions récentes des bus d'une ligne spécifique
     *
     * @param ligne Numéro/Nom de la ligne de bus
     * @param minutes Nombre de minutes dans le passé à considérer
     * @return Liste des positions récentes
     */
    List<BusPosition> getRecentBusPositions(String ligne, int minutes);

    /**
     * Récupère la dernière position connue d'un bus spécifique
     *
     * @param busId Identifiant du bus
     * @return Dernière position connue ou null si aucune position n'est trouvée
     */
    BusPosition getLastBusPosition(String busId);
}

File: geobusandroid/app/src/main/java/com/geobus/marrakech/network/BusPositionService.kt
package com.geobus.marrakech.network

import com.geobus.marrakech.model.BusPosition
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

/**
 * Interface pour les requêtes liées aux positions des bus
 */
interface BusPositionService {
    
    /**
     * Récupère toutes les dernières positions des bus
     * 
     * @return Liste des positions des bus
     */
    @GET("positions")
    suspend fun getAllLatestPositions(): List<BusPosition>
    
    /**
     * Récupère les bus qui se dirigent vers un arrêt spécifique
     * 
     * @param stopId Identifiant de l'arrêt
     * @return Liste des positions des bus allant vers cet arrêt
     */
    @GET("positions/stop/{stopId}")
    suspend fun getBusesGoingToStop(@Path("stopId") stopId: Long): List<BusPosition>
    
    /**
     * Ajoute une nouvelle position de bus
     * 
     * @param busPosition Position du bus à ajouter
     * @return Position du bus enregistrée
     */
    @POST("positions")
    suspend fun addBusPosition(@Body busPosition: BusPosition): BusPosition
}